pmd-core/src/main/java/net/sourceforge/pmd/lang/metrics/MetricKey.java

```diff
@@ -44,4 +51,53 @@
 
     // TODO the metric key should know about supported options
 
+
+    /**
+     * Creates a new metric key from its metric and name.
+     *
+     * @param name   The name of the metric
+     * @param metric The metric to use
+     * @param <T>    Type of node the metric can be computed on
+     *
+     * @return The metric key
+     *
+     * @throws NullPointerException If either parameter is null
+     */
+    static <T extends Node> MetricKey<T> of(@NonNull String name, @NonNull Metric<T> metric) {
```

Comment:
Actually, do we still need this factory method? On master, I don't see any usage of `MetricKeyUtil.of(...)` other than in tests... Didn't look into the 7.0.0 branch yet

Reply:
It's true that it's only used in tests for now, but he doc on the website mentions that this is meant to be a simple extension hook for users defining their own metrics. I think it comes down to the question: is this a use case we really want to support? I would think yes, because it's easy to support. OTOH I think the framework is likely to change more for 7.0 (because eg, it's weird to restrict metrics to two types of nodes per language), and maybe this extension mechanism will be broken.

I think the safest thing for now is to keep it until we know more about what the framework looks like. I'll mention this on the wiki

Reply:
:+1: